gax-grpc/src/main/java/com/google/api/gax/grpc/UnaryCallable.java

```diff
@@ -328,4 +329,8 @@ public ResponseT call(RequestT request) {
     return new UnaryCallable<>(
         new BatchingCallable<>(callable, batchingDescriptor, batcherFactory), channel, settings);
   }
+
+  UnaryCallable<RequestT, ResponseT> auth(CallCredentials credentials) {
```

Comment:
I think the main reason we should not directly use the provider is that it's ambiguous who, or whether, the result is memoized. We probably shouldn't read the environment for default credentials on every call, etc.